### PR TITLE
feat: Table supports sort state

### PR DESCRIPTION
**Pull request by ling1726**
Adds sort state to the `Table` component.

Addresses #23983

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/26e5a29ec185ffaf62407b76e63190a4adb8e012'>26e5a29ec185ffaf62407b76e63190a4adb8e012</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-table/src/stories/Table/Sort.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    author: { label: 'Jane Doe', status: 'offline' },
+    lastUpdated: { label: 'Tue at 9:30 AM', timestamp: 1 },
+    lastUpdate: {
+      label: 'You shared this in a Teams chat',
+      icon: <PeopleRegular />,
+    },
+  },
+];
+
+const columns: Record<string, { label: string; compare: (a: unknown, b: unknown) => number }> = {
```
Comment (ling1726) [inline packages/react-components/react-table/src/stories/Table/Sort.stories.tsx:83]: used `unknown` here for typing, any suggestions welcome

**Review by bsunderhus — APPROVED** [review]
Little bit different, but since both values are definitely intertwined, makes sense! LGTM! 👍🏼

**Comment by ling1726** [comment]
> Little bit different, but since both values are definitely intertwined, makes sense! LGTM! 👍🏼

Main issue is `useControllableState` if I controlled each of those values separately it risks UI tearing (i.e. sort direction and column aren't in sync) and I don't want to open the can of worms that `useControllableReducer` would be 😱😱

**Comment by bsunderhus** [comment]
>  I don't want to open the can of worms that `useControllableReducer` would be 😱😱

You got my attention there 👀

**Pull request merged**